flo-freezer/src/main/java/com/spotify/flo/freezer/PersistingContext.java

```diff
@@ -95,39 +97,48 @@ public PersistingContext(Path basePath, EvalContext delegate) {
     return promise.value();
   }
 
-  public static void serialize(Object object, Path file) throws Exception{
-    final Kryo kryo = new Kryo();
-    PackageRegistrar.all().apply(kryo);
-    kryo.register(java.lang.invoke.SerializedLambda.class);
-    kryo.register(ClosureSerializer.Closure.class, new ClosureSerializer());
-    kryo.addDefaultSerializer(java.lang.Throwable.class, new JavaSerializer());
-    kryo.getFieldSerializerConfig().setIgnoreSyntheticFields(false);
-
-    if (Files.exists(file)) {
-      throw new RuntimeException("File " + file + " already exists");
-    }
-
+  public static void serialize(Object object, Path file) throws Exception {
     try {
-      try (Output output = new Output(newOutputStream(file, WRITE, CREATE_NEW))) {
-        kryo.writeClassAndObject(output, object);
-      }
+      serialize(object, Files.newOutputStream(file, WRITE, CREATE_NEW));
     } catch (IOException e) {
       throw new RuntimeException(e);
     }
   }
 
+  public static void serialize(Object object, OutputStream outputStream) {
+    final Kryo kryo = getKryo();
+
+    try (Output output = new Output(outputStream)) {
+      kryo.writeClassAndObject(output, object);
+    }
+  }
+
   public static <T> T deserialize(Path filePath) throws Exception {
+    return deserialize(Files.newInputStream(filePath));
+  }
+
+  public static <T> T deserialize(InputStream inputStream) {
+    final Kryo kryo = getKryo();
+
+    try (Input input = new Input(inputStream)) {
+      return (T) kryo.readClassAndObject(input);
+    }
+  }
+
+  private static Kryo getKryo() {
     Kryo kryo = new Kryo();
     PackageRegistrar.all().apply(kryo);
     kryo.register(java.lang.invoke.SerializedLambda.class);
+    try {
+      kryo.register(Class.forName("com.typesafe.config.impl.SimpleConfig"), new ConfigSerializer());
+    } catch (ClassNotFoundException e) {
+      LOG.debug("ConfigSerializer not registered", e);
+    }
     kryo.register(ClosureSerializer.Closure.class, new ClosureSerializer());
     kryo.setInstantiatorStrategy(new Kryo.DefaultInstantiatorStrategy(new StdInstantiatorStrategy()));
```

Comment:
It seems we don't set this for serializing. What does it achieve?

Reply:
I am not sure either. I am trying to deduplicate the code a bit. Should we go back having that part duplicated?

Reply:
We will take a look later to make sure we don't break anything.

Reply:
It doesn't seem to matter because it is only used for deserializing, just being weird. We might do that after https://github.com/spotify/flo/pull/95/files#diff-d7f1d9c331c8fb2c839cdd4ff6cb0a16R121 to make it less confusing.

Reply:
Sure. Shall I do the change?

Reply:
Sure please go ahead. You own this PR. =D

Reply:
👍  Done.